src/condvar.rs

```diff
@@ -387,12 +387,17 @@ impl Condvar {
     ///
     /// Like `wait`, the lock specified will be re-acquired when this function
     /// returns, regardless of whether the timeout elapsed or not.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the given `timeout` is so large that it can't be added to the current time.
     #[inline]
     pub fn wait_for<T: ?Sized>(
         &self,
         guard: &mut MutexGuard<T>,
         timeout: Duration,
     ) -> WaitTimeoutResult {
+        // FIXME: Change to Intstant::now().checked_add(timeout) when stable.
```

Comment:
Should this `FIXME` also be added to `raw_mutex.rs` and `raw_rwlock.rs` where appropriate?

Reply:
Good idea. Fixed.